ISPy/util/transformation.py

```diff
@@ -0,0 +1,237 @@
+import numpy as np
+from astropy.io import fits
+import astropy.units as u
+import interpolate2d
+
+
+# ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+def sphere2img(lat, lon, latc, lonc, xcenter, ycenter, rsun, peff):
+    """Conversion between Heliographic coordinates to CCD coordinates.
+    Ported from sphere2img written in IDL : Adapted from Cartography.c by Rick Bogart, 
+    by Xudong Sun [Eq 5&6 in https://arxiv.org/pdf/1309.2392.pdf]
+
+    Parameters
+    ----------
+    lat, lon : array, array
+        input heliographic coordinates (latitude and longitude)
+    latc, lonc : float, float
+        Heliographic longitud and latitude of the refenrence (center) pixel
+    xcenter, ycenter : float, float
+        Center coordinates in the image
+    rsun : float
+        Solar radius in pixels
+    peff : [type]
+        p-angle (CW rotation on the image nneded for the solar north to be +eta)
+
+    Returns
+    -------
+    array
+        Latitude and longitude in the new CCD coordinate system.
+    """
+    sin_asd = 0.004660
+    cos_asd = 0.99998914
+    last_latc = 0.0
+    cos_latc = 1.0
+    sin_latc = 0.0
+
+    if (latc != last_latc):
+        sin_latc = np.sin(latc)
+        cos_latc = np.cos(latc)
+        last_latc = latc
+
+    sin_lat = np.sin(lat)
+    cos_lat = np.cos(lat)
+    cos_lat_lon = cos_lat * np.cos(lon - lonc)
+
+    cos_cang = sin_lat * sin_latc + cos_latc * cos_lat_lon
+    r = rsun * cos_asd / (1.0 - cos_cang * sin_asd)
+    xr = r * cos_lat * np.sin(lon - lonc)
+    yr = r * (sin_lat * cos_latc - sin_latc * cos_lat_lon)
+
+    cospa = np.cos(peff)
+    sinpa = np.sin(peff)
+    xi = xr * cospa - yr * sinpa
+    eta = xr * sinpa + yr * cospa
+
+    xi = xi + xcenter
+    eta = eta + ycenter
+
+    return xi, eta
+
+
+
+# ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+def vector_transformation(peff,latitude_out,longitude_out, B0,field_x_cea,
+        field_y_cea, field_z_cea, bvec2cea=False):
+    """
+    Magnetic field transformation matrix (see Allen Gary & Hagyard 1990)
+    [Eq 7 in https://arxiv.org/pdf/1309.2392.pdf]
+    """
+
+    nlat_out = len(latitude_out)
+    nlon_out = len(longitude_out)
+
+    PP = peff
+    if bvec2cea is False:
+        BB = latitude_out[None, 0:nlat_out] * np.pi / 180.0
+        LL = longitude_out[0:nlon_out, None] * np.pi / 180.0
+    else:
+        BB = latitude_out
+        LL = longitude_out
```

Comment:
When `bvec2cea` is `False` we are still not really doing `bcea2vec` (which is what it implies to me). Maybe rename the keyword?

Reply:
Fixed